src/viz/Ensight_Stream.cc

```diff
@@ -71,19 +73,33 @@ Ensight_Stream::~Ensight_Stream(void) { close(); }
  * \param file_name  Name of output file.
  * \param binary     If true, output binary.  Otherwise, output ascii.
  * \param geom_file  If true, then a geometry file will be dumped.
+ * \param decomposed If true, input is domain decomposed. Otherwise domain replicated.
  */
 void Ensight_Stream::open(const std::string &file_name, const bool binary,
-                          const bool geom_file) {
+                          const bool geom_file, const bool decomposed) {
   Require(!file_name.empty());
 
   d_binary = binary;
 
   // Open the stream.
-  if (binary)
-    d_stream.open(file_name.c_str(), std::ios::binary);
-  else
-    d_stream.open(file_name.c_str());
-
+  if (decomposed) {
+    if (binary)
+      d_decomposed_stream.reset(
+          new rtt_c4::ofpstream(file_name, std::ios::binary));
+    else
+      d_decomposed_stream.reset(new rtt_c4::ofpstream(file_name));
+    // set to a generic ostream
+    d_stream = &*d_decomposed_stream;
+  } else {
+    Insist(rtt_c4::node() == 0, "Ensight_Stream, called by nonzero rank with "
+                                "the domain decomposed flag");
```

Comment:
Is this error message accurate? I thought this threw because a nonzero rank was calling Ensight Stream withOUT the domain-decomposed flag :)

Reply:
Good catch. I'll fix this.